File: steit-csharp-v2/src/Codec/WireType.cs
using System;

namespace Steit.Codec {
    public enum WireType : Byte {
        Varint = 0,
        Sized = 2,
    }

    public static class WireTypeHelper {
        public static WireType New(UInt32 value) {
            switch (value) {
                case (byte) WireType.Varint:
                case (byte) WireType.Sized:
                    return (WireType) value;

                default:
                    throw new ArgumentException(String.Format("Invalid wire type value: {0}", value));
            }
        }
    }
}

File: steit-go/pkg/codec/wiretype.go
package codec

import "fmt"

type WireType = int

const (
	WireTypeSized  WireType = 2
	WireTypeVarint WireType = 0
)

func GetWireTypePtr(wireType WireType) *WireType {
	return &wireType
}

func NewWireType(value uint32) WireType {
	switch value {
	case uint32(WireTypeVarint):
		return WireType(value)
	case uint32(WireTypeSized):
		return WireType(value)
	default:
		panic(fmt.Sprintf("Invalid wire type value: %d", value))
	}
}
